galsim/base.py

```diff
@@ -2892,6 +2892,119 @@ def __str__(self):
 _galsim.SBSpergel.__setstate__ = lambda self, state: 1
 _galsim.SBSpergel.__repr__ = lambda self: \
         'galsim._galsim.SBSpergel(%r, %r, %r, %r, %r)'%self.__getinitargs__()
+    
+
+class InclinedExponential(GSObject):
+    """A class describing an inclined exponential profile.
+
+    The Inclined Exponential surface brightness profile is characterized by three properties: its
+    inclination angle 'i' (where 0 = face-on and pi/2 = edge-on), its scale radius, and its scale
+    height. The 3D light distribution function is:
+    
+    I(R,z) = I_0 / (2h_s) * sech^2 (z/h_s) * exp(-R/R_s)
+    
+    where z is the distance along the minor axis, R is the radial distance from the minor axis,
+    R_s is the scale radius of the disk, h_s is the scale height of the disk, and I_0 is the central
+    surface brightness of the face-on disk.
+    
+    In this implementation, the profile is inclined along the y-axis. This means that it will likely
+    need to be rotated in most circumstances.
+    
+    At present, this profile is not enabled for photon-shooting.
+    
+    Initialization
+    --------------
+
+    @param i                The inclination angle, i
+    @param scale_radius     The scale radius of the exponential disk.  Typically given in arcsec.
+    @param scale_height     The scale height of the exponential disk.  Typically given in arcsec.
+    @param flux             The flux (in photons) of the profile. [default: 1]
+    @param gsparams         An optional GSParams argument.  See the docstring for GSParams for
+                            details. [default: None]
+
+    Methods
+    -------
+
+    In addition to the usual GSObject methods, InclinedExponential has the following access methods:
+
+        >>> i = inclined_exponential_obj.getI()
+        >>> r0 = inclined_exponential_obj.getScaleRadius()
+        >>> h0 = inclined_exponential_obj.getScaleHeight()
+    """
+    _req_params = { "i" : float, "scale_radius" : float, "scale_height" : float }
+    _opt_params = { "flux" : float }
+    _takes_rng = False
+
+    def __init__(self, i, scale_radius, scale_height,
+                 flux=1., gsparams=None):
+        
+        if not isinstance(i, galsim.Angle):
+            raise TypeError("Input i should be an Angle")
+        
+        i_rad = i.rad()
```

Comment:
There's no need for this.  `Angle`s are python-wrapped, so you should be able to pass in the `Angle` object to C++ and use it as such there and keep the unit safety afforded by the `Angle` class.